warehouse/forklift/metadata.py

```diff
@@ -235,6 +236,19 @@ def _validate_metadata(metadata: Metadata, *, backfill: bool = False):
                         )
                     )
 
+    # Validate that any `dynamic` fields passed are in the allowed list
+    # TODO: This probably should be lifted up to packaging.metadata
+    for field in {"dynamic"}:
+        if (value := getattr(metadata, field)) is not None:
+            for key in value:
+                if key not in map(str.lower, DynamicFieldsEnum.enums):
```

Comment:
will `key` already have been normalized to lowercase for comparison?

Reply:
I had to check, but yes - packaging does that for us
https://github.com/pypa/packaging/blob/9375ec2eff48257967c97d331b9a76019e95bdb0/src/packaging/metadata.py#L314-L316

Reply:
I added a test case change to prove the case-insensitive behavior, just in case.

Reply:
Thank you!